casanovo/denovo/model.py

```
@@ -1493,51 +1501,60 @@ def get_lr_factor(self, epoch):
+        aa_scores = torch.as_tensor(aa_scores)
 
-        return peptide_score
+    eps = torch.finfo(torch.float64).eps
+    log_scores = torch.log(torch.clamp(aa_scores.double(), eps, 1))
 
-    # BATCH PATH: database search
+    if aa_scores.ndim == 1:
+        # FAST PATH: de novo inference — single peptide.
+        peptide_score = torch.exp(torch.sum(log_scores))
+        return peptide_score.item() if return_numpy else peptide_score
     else:
```

Ruff (0.15.17)

[warning] Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)